docs/implementation.md

```diff
@@ -32,3 +32,14 @@ There are two primary entry points for middleware:
 
 1. `robot.listenerMiddleware` - registers a new piece of middleware in a global array
 2. `middleware.execute` - executes all registered middleware in order
+
+## Persistence
+
+Hubot has an in-memory key-value store exposed as `robot.brain` that can be used to store and retrieve data.
+Furthermore, `hubot-redis-brain` enables persistence across Hubot restarts.
+This package requires a Redis server.
+
+By default, the brain contains a list of all users seen by Hubot.
+Therefore, without persistence across restarts, the brain will contain the list of users encountered so far, during the current run of Hubot.
+On the other hand, with persistence across restarts, the brain will contain all users encountered by Hubot during all of its runs.
+This list of users can be accessed through `hubot.brain.users` and [other utility methods](https://github.com/github/hubot/blob/master/src/brain.coffee#L94-L136).
```

Comment:
`hubot.brain.users` is actually [a function](https://github.com/github/hubot/blob/master/src/brain.coffee#L94-L98), so I'd include parens on it ie `hubot.brain.users()`.

I also wouldn't link to line numbers in brain.coffee in documentation either, because that is a good way for it to become out of date.